src/thread.rs

```diff
@@ -212,6 +414,11 @@ pub struct ThreadManager<'mir, 'tcx> {
     ///
     /// Note that this vector also contains terminated threads.
     threads: IndexVec<ThreadId, Thread<'mir, 'tcx>>,
+    /// Set of threads that are currently yielding.
```

Comment:
"Yield" seems like an atomic operation to me, it's what happens on a single call of `yield_now`... do you mean threads that are currently recording a watch set?

Reply:
Is there some kind of invariant that all threads in this set are in state `DelayOnYield` or so?